src/Symfony/Component/Cache/Traits/RedisClusterNodeProxy.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Traits;
+
+use \RedisCluster;
+
+/**
+ * This file acts as a wrapper to the RedisCluster implementation
+ *
+ * Calls are made to nodes via RedisCluster->{method}($host, ...args)'
+ *  according to https://github.com/phpredis/phpredis/blob/develop/cluster.markdown#directed-node-commands
+ *
+ * @author Jack Thomas <jack.thomas@solidalpha.com>
+ *
+ * @internal
+ */
+class RedisClusterNodeProxy
+{
+    /** @var array */
+    private $host;
+
+    private $redis;
+
+    /**
+     * @param array $redisHost
+     * @param \RedisCluster|RedisClusterProxy $redis
+     */
+    public function __construct(array $host, $redis)
+    {
+        $this->host = $host;
+        $this->redis = $redis;
+
+        /*
+        Old Implementation:
+        use \Redis;
+
+        $h = new Redis();
+        $h->connect($host[0], $host[1]);
+        */
+    }
+
+    public function __call(string $method, array $args)
+    {
+        return $this->redis->{$method}($this->host, ...$args);
+    }
+
+    public function scan(&$iIterator, $strPattern = null, $iCount = null)
+    {
+        return $this->redis->scan($iIterator, $this->host, $strPattern, $iCount);
```

Comment:
this one is strange, can you double check that `$iIterator` comes before the host?

Reply:
the doc linked at the top of that class says that the host is always the first argument. So this indeed looks strange

Reply:
Annoyingly, the upstream driver implements scan with iterator first for its scan method.

https://github.com/phpredis/phpredis/blob/develop/redis_cluster.c#L92

Reply:
👍 ?

Reply:
@jacekwilczynski then this means that the upstream documentation (the one you linked in the phpdoc of the class) is wrong when saying that the host is passed as the first argument. I suggest you to contribute upstream to fix that doc.